### PR TITLE
New FTP-commands

### PR DESCRIPTION
**Pull request by akoro**
Hello, Peter! I wrote to you two weeks ago.
1) I have added MLSD, NOOP, OPTS and FEAT commands. It is need for Win10 FTP and for TotalCommander FTP-connection.
2) The system crashed and rebooted when accidentally attempting to execute CDUP in the root directory. Now I have fixed it too.
I hope this improves the library.

### CODE REVIEW

File: src/Commands/LIST.h [hunk 1]
```diff
@@ -11,22 +11,23 @@ class LIST : public FTPCommand
...
-	
+
 	void run(FTPPath & WorkDirectory, const std::vector<String> & Line) override
 	{
 		if(!ConnectDataConnection())
 		{
 			return;
 		}
-		File dir = _Filesystem->open(WorkDirectory.getPath());
+		File dir = _Filesystem->open(WorkDirectory.getPath()); //
```
Comment (peterus) [inline src/Commands/LIST.h:21]: please remove comment symbols

File: src/Commands/MLSD.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      data_println(filename);
+
+      cnt++;
+      f = root.openNextFile();
+    }
+
+    root.close();
+    CloseDataConnection();
+    SendResponse(226, "MLSD completed");
+//    SendResponse(226, String(cnt) + " matches total");
```
Comment (peterus) [inline src/Commands/MLSD.h:83]: please remove line if not needed

File: src/FTPFilesystem.h [hunk 3]
```diff
@@ -15,7 +15,7 @@ class FTPFilesystem
 	void addFilesystem(String Name, FS * const Filesystem);
 	void clearFilesystemList();
 
-	File open(const String & path, const char* mode = FILE_READ);
+	File open(const String & path, const char* mode = "r");
```
Comment (peterus) [inline src/FTPFilesystem.h:18]: is the `FILE_READ` not supported on ESP8266? I would like to stay with constants if possible.

File: src/FTPPath.cpp [hunk 4]
```diff
@@ -16,13 +16,21 @@ FTPPath::~FTPPath()
 
 void FTPPath::changePath(String path)
 {
-	std::list<String> p = splitPath(path);
-	std::copy(p.begin(), p.end(), std::back_inserter(_Path));
+if(*path.begin() == '/') // Added Akoro 2021-02-27
```
Comment (peterus) [inline src/FTPPath.cpp:19]: please use `path.StartsWith()`: https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/startswith/

**Review by peterus — CHANGES_REQUESTED** [review]
thanks for your contribution! i really like this changes!
I made some comments, please fix this small requests and I will be happy to accept your PR 👍

File: src/FTPPath.cpp (hunk 4 above)
Comment (akoro) [reply src/FTPPath.cpp]: It is don't need. I have returned to originnal code

File: src/FTPFilesystem.h (hunk 3 above)
Comment (akoro) [reply src/FTPFilesystem.h]: Unfortunately it is. I return constant

File: src/Commands/MLSD.h (hunk 2 above)
Comment (akoro) [reply src/Commands/MLSD.h]: OK

File: src/Commands/LIST.h (hunk 1 above)
Comment (akoro) [reply src/Commands/LIST.h:21]: OK

**Comment by peterus** [comment]
next time you can just push new changes to the same repo and this changes will aper in this pull request ;) you don't need to create another pull request. thanks for this fixes again!

**Pull request closed without merging**